openedx/core/djangoapps/user_api/management/commands/sync_hubspot_contacts.py

```diff
@@ -41,57 +40,44 @@ def _get_hubspot_enabled_sites(self):
         ]
         return hubspot_sites
 
-    def _get_last_synced_contact_email(self, site_conf):
+    def _get_users_queryset(self, initial_days):
         """
-        Returns: last synced contact email for given site
-
+        initial_days: numbers of days to go back from today
+        :return: users queryset
         """
-        api_key = site_conf.get_value('HUBSPOT_API_KEY')
-        last_contact_email = None
-        client = EdxRestApiClient(urlparse.urljoin(HUBSPOT_API_BASE_URL, 'contacts/v1/lists/all/contacts'))
-        try:
-            response = client.recent.get(hapikey=api_key, count=1, property='email')
-            if 'contacts' in response:
-                for contact in response['contacts']:
-                    last_contact_email = contact.get('properties').get('email').get('value')
-
-        except (HttpClientError, HttpServerError) as ex:
-            message = u'An error occurred while getting recent contact for site {domain}, {message}'.format(
-                domain=site_conf.site.domain, message=ex.message
-            )
-            self.stderr.write(message)
-        return last_contact_email
+        start_date = datetime.now().date() - timedelta(initial_days)
+        end_date = datetime.now().date() - timedelta(1)
```

Comment:
with `timedelta(1)` I believe this would only grab users up until the previous day? This also means if `initial_days=1` no users would captured. I find this confusing giving the method comment and think it would make more sense if end_date was just `datetime.now().date()`  Was there a reason to stop at yesterday?

Reply:
After executing line `end_date = datetime.now().date() - timedelta(1)` 
`end_date` will hold previous date in it.
and if we set `initial_days=1` then `start_date` will also hold previous date.
this means our query 
```
users_qs = User.objects.filter(
            date_joined__date__gte=start_date,
            date_joined__date__lte=end_date
        ).order_by('id')
```
will capture all users that have joined on previous day.

We are syncing users till previous date to avoid any possible repetition or inconsistency.

Reply:
Okay - thank you for the clarification